### PR TITLE
560 eav attribute generation refactoring

### PR DESCRIPTION
**Pull request by ProkopovVitaliy**
**Description** (*)

Refactored dialogs hierarchy for eav attribute generation.

Separated generation actions for different entities:
Screenshot 2021-04-19 at 08 49 27

**Fixed Issues (if relevant)**

1. Fixes magento/magento2-phpstorm-plugin#560

**Questions or comments**

**Contribution checklist** (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with integration/functional tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/NewProductEavAttributeDialog.java [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    @FieldValidation(rule = RuleRegistry.COMMA_SEPARATED_STRING,
+            message = {CommaSeparatedStringRule.MESSAGE, "Apply To"})
+    private JCheckBox applyToAllProductsCheckBox;
+    private JPanel applyToPanel;
+    private JList productsTypesList;
+
+    /**
+     * Constructor.
+     *
+     * @param project   Project
```
Comment (VitaliyBoyko) [inline src/com/magento/idea/magento2plugin/actions/generation/dialog/NewProductEavAttributeDialog.java:96]:
```suggestion
     * @param project Project
```

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/event/eavdialog/SourceModelNameAdapter.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    private void updateSourceModelName(final String attributeCode) {
+        if (attributeCode.isEmpty()) {
+            sourceModelNameTexField.setText("");
+
+            return;
+        }
+
+        final String[] attributeCodeParts = attributeCode.split("_");
```
Comment (VitaliyBoyko) [inline src/com/magento/idea/magento2plugin/actions/generation/dialog/event/eavdialog/SourceModelNameAdapter.java:45]: "_" - this literal is duplicated several times. Could you please use a constant?

File: src/com/magento/idea/magento2plugin/actions/groups/NewEavAttributeGroup.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import com.intellij.ide.actions.NonTrivialActionGroup;
+import com.intellij.openapi.util.IconLoader;
+import com.magento.idea.magento2plugin.MagentoIcons;
+import javax.swing.Icon;
+import org.jetbrains.annotations.NotNull;
+
+public class NewEavAttributeGroup extends NonTrivialActionGroup {
+
+    /**
+     * Group for eav attribute generation actions.
```
Comment (VitaliyBoyko) [inline src/com/magento/idea/magento2plugin/actions/groups/NewEavAttributeGroup.java:17]:
```suggestion
     * Group for EAV attribute generation actions.
```

**Review by VitaliyBoyko — CHANGES_REQUESTED** [review]

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/event/eavdialog/SourceModelNameAdapter.java (hunk 2 above)
Comment (ProkopovVitaliy) [reply src/com/magento/idea/magento2plugin/actions/generation/dialog/event/eavdialog/SourceModelNameAdapter.java]: @VitaliyBoyko Created specialised Util for this aim.
com.magento.idea.magento2plugin.actions.generation.dialog.util.EavAttributeCodeSeparatorUtil

Thanks!

**Comment by ProkopovVitaliy** [comment]
@VitaliyBoyko Requested changes implemented. 

Thanks!

**Review by VitaliyBoyko — APPROVED** [review]

**Pull request merged**